src/components/led/led_strip_6812_rgbww.py

```diff
@@ -18,40 +18,35 @@ def __init__(self, lamp, color, pin, num_pixels):
 
     # Turn this LED Strip off
     def off(self):
-        self.pixels.fill((0,0,0,0))
-        self.pixels.write()
-
-    # Reset to default
-    def reset(self):
-        self.update_colors(self.default_pixels)
+        self.fill((0,0,0,0))
 
-   # set to a new color (tuple of rgbw color or list of individual pixels)
+    # Set to a new color (tuple of rgbw color or list of individual pixels)
     def fill(self, color):
-        if isinstance(color, list): 
-            for i in range(self.num_pixels): 
+        if isinstance(color, list):
+            for i in range(self.num_pixels):
                 self.pixels[i] = color[i]
-        else: 
+        else:
             self.pixels.fill(color)
 
         self.pixels.write()
 
     # Shift pixels from their current state to a target state. Dest can be either a list of individual pixels or a RGBW tuple
-    # :TODO: Allow easing type to be passed
-    async def async_fade(self, dest, steps, step_delay=1):
-        if not isinstance(dest, list): dest = [dest] * self.num_pixels
+    async def fade(self, dest, steps, step_delay=1, easing_function=QuadEaseInOut):
```

Comment:
The reason this was called async_fade before is beacuse it backgrounds , intended for fades that take a long time and can be blocked by other things

It's probably right to rename it like you did, we just may want in the future to give it an option to block while fading for sync fades or something, where it skips the lock check 

Basically where the lock is now it will lock for each step, but also each step that means if something else is locking this will pause and wait (good example is gramp - if it's doing a 5 minute long fade and you touch the base, the lamp dims, the fade stops until you let go and then resumes)

Reply:
I think this change was a result of a TODO in gramp.py. I think a solid explanation of the locks on the led strips to users is a great idea to add to our docs

Reply:
https://github.com/Lamprichauns/lamp-os/issues/31